File: 백준/Silver/10828. 스택/스택.py
import sys
N = int(sys.stdin.readline())

arr=[]
for i in range(N):
    word = sys.stdin.readline().split()
    if(word[0]=='push'):
        arr.append(word[1])
    if(word[0]=='top'):
       if(len(arr)==0) : print(-1) 
       else : print(arr[-1])
    if(word[0]=='size'):
        print(len(arr))
    if(word[0]=='empty'):
        if(len(arr)==0): print(1)
        else :print(0)
    if(word[0]=='pop'):
          if(len(arr)==0): print(-1)
          else : print(arr.pop())            
        

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException{  
        try (BufferedReader bf = new BufferedReader(new InputStreamReader(System.in))) {
            int N = Integer.parseInt(bf.readLine()); 
            ArrayList<String[]> commend = new ArrayList<>();
            for(int i=0;i<N;i++){
                String[] word = bf.readLine().split(" ");
                commend.add(word);
            }
            Stack<String> arr = new Stack<>();
            for(String[] word : commend){
                if(word[0].equals("push")){
                    arr.push(word[1]);
                }
                if(word[0].equals("top")){
                    if(arr.isEmpty()) System.out.println(-1);
                    else System.out.println( arr.peek()); 
                }
                if(word[0].equals("pop")){
                    if(arr.isEmpty()) System.out.println(-1);
                    else System.out.println(arr.pop()); 
                }
                if(word[0].equals("size")){
                     System.out.println(arr.size());
                }
                if(word[0].equals("empty")){
                     if(arr.empty())  System.out.println(1);
                     else  System.out.println(0);
                }  
            }
        }
    }
}
